### PR TITLE
Sphinx documentation generation in travis (#4).

### PR DESCRIPTION
**Pull request by imagovrn**
- Bumped down required CMake version back to 3.5.
- Optional build of docs if CMake 3.9+ is present.
- Building documentation in Travis.
- Updated travis file: variable for docker image,
  mounted docker build, preparation for docs deployment.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,18 +4,26 @@ language: cpp
...
   - |
     docker run -t \
       -e C_COMPILER=${C_COMPILER} \
       -e CXX_COMPILER=${CXX_COMPILER} \
       -v $(pwd):/src \
+      -v $(pwd)/build:/build \
       -w /build \
-      boxdot/flatdata-dev /src/ci/build-and-test.sh
+      $DOCKER_IMAGE /src/ci/build-and-test.sh
+  - sudo chown -R travis:travis $(pwd)/build
```
Comment (boxdot) [inline .travis.yml]: Are you sure this will work? You are not saying sudo required in travis.

File: .travis.yml [hunk 2]
```diff
@@ -4,18 +4,26 @@ language: cpp
...
+
+before_deploy:
+  - touch $(pwd)/build/docs/html/.nojekyll
```
Comment (boxdot) [inline .travis.yml]: I am confused, why there is `before_deploy` but no `deploy` directive.

File: .travis.yml (hunk 1 above)
Comment (imagovrn) [reply .travis.yml]: It does. Tried it in my repo + verification passed. Maybe it works by chance? Should I state something extra?

File: .travis.yml (hunk 2 above)
Comment (imagovrn) [reply .travis.yml]: I kept "before deploy" so the docs preparation is captured. Deploy can't be done just yet. I can also remove it if you like.

**Comment by imagovrn** [comment]
@boxdot @VeaaC should we submit or should I change anything? :)

**Comment by boxdot** [comment]
Any progress on that? Or should we abandon it?

**Pull request closed without merging**